ui/src/data_explorer/containers/DataExplorer.js

```diff
@@ -79,15 +84,24 @@ const DataExplorer = React.createClass({
       queryConfigs,
       queryConfigActions,
       source,
+      hideWriteFormAction,
+      showWriteFormAction,
+      dataExplorer: {showWriteForm},
     } = this.props
     const {activeQueryIndex} = this.state
 
     return (
       <div className="data-explorer">
+        {showWriteForm
+          ? <OverlayTechnologies>
+              <WriteDataForm onClose={hideWriteFormAction} source={source} />
+            </OverlayTechnologies>
+          : null}
         <Header
           actions={{handleChooseAutoRefresh, setTimeRange}}
           autoRefresh={autoRefresh}
           timeRange={timeRange}
+          showWriteFormAction={showWriteFormAction}
```

Comment:
i'd suggest passing these through as `showWriteForm` (or possibly even `onShowWriteForm` -- your call), per my below comment, unless there's a namespace collision at either this component or in `<Header>`